### PR TITLE
Support the same stemming logic when coming in from top level vs. a L…

### PR DESCRIPTION
**Pull request by tamarahills**
…aunch intent.
Also I fixed few cases where when you say "Alexa Stop" the state is not getting properly reset to START due to that API bug.

### CODE REVIEW

**Comment by tamarahills** [comment]
This fixes #6

File: state_handlers.js [hunk 1]
```diff
@@ -352,6 +361,37 @@ var scout_agent = (function() {
...
+      stateObjThis.attributes['titles'] = titles;
+      stateObjThis.handler.state = constants.states.TITLES_DECISION_MODE;
+      matchArticleToTitlesHelper(stateObjThis);
+    });
+  } else {
+    stateObj.handler.state = constants.states.TITLES_DECISION_MODE;
+    matchArticleToTitlesHelper(stateObj);
+  }
+}
+
```
Comment (dylano) [inline state_handlers.js]: Suggestion from Slack thread:
image

File: state_handlers.js [hunk 2]
```diff
@@ -214,7 +199,12 @@ var state_handlers = {
       },
       'AMAZON.StopIntent': function() {
         console.log('TITLES_DECISION_MODE:AMAZON.StopIntent');
+        this.handler.state = constants.states.START_MODE;
         this.response.speak(constants.strings.ALEXA_STOP_RESP);
+        this.handler.state = '';
```
[the remark this answers is not in the document]
Comment (tamarahills) [reply state_handlers.js]: Yes, there is a whole thread on the forums about this.  I think this combination seems to work.

File: state_handlers.js (hunk 1 above)
Comment (tamarahills) [reply state_handlers.js]: Good suggestion. Changing.  I tested, it works.

**Pull request merged**